extensions/amp-analytics/0.1/requests.js

```diff
@@ -428,3 +481,20 @@ function expandExtraUrlParams(
 
   return Promise.all(requestPromises).then(() => params);
 }
+
+/**
+ * Tries to combine the given domain and relative url
+ * @param {string} relativeUrl
+ * @param {string} baseUrl
+ * @return {string|undefined}
+ */
+function resolveRelativeUrlString_(relativeUrl, baseUrl) {
+  const baseDomain = extractDomain(baseUrl);
+  const relativePath = extractRelativePath(relativeUrl);
+
+  if (baseDomain && relativePath) {
+    return baseDomain + relativePath;
+  }
+
+  return undefined;
```

Comment:
what happens if we return undefined? Should this be a developer or user error?

Reply:
Since this is a helper function, I was thinking of handling the error logic outside of this function.

When we return undefined it means that the given URLs are invalid so we should throw a user error for them to fix the URLs in the config.